services/pixiv.py

```diff
@@ -0,0 +1,136 @@
+import time
+
+import pixivpy3 as pixiv
+from gppt import GetPixivToken
+import sched
+import threading
+
+
+def singleton(cls):
+    instances = {}
+
+    def get_instance(*args, **kwargs):
+        if cls not in instances:
+            instances[cls] = cls(*args, **kwargs)
+        return instances[cls]
+
+    return get_instance
+
+
+def packIllustUrl(uid) -> str:
+    return "https://www.pixiv.net/artworks/%s" % uid
+
+
+@singleton
+class Pixiv:
+
+    def __init__(self, username, password, retry=5, interval=3599):
```

Comment:
def __init__(self, username: str, password: str, retry: int = 5, interval: int = 3599):
        """
        Start the Pixiv service for tg_bot.

        :param username: Given Pixiv username.
        :param password: Given Pixiv password.
        :param retry: Maximum retry times.
        :param interval: Auto-refresh interval for the Pixiv token.
        """

Reply:
Can can need